workspaces/util/src/__tests__/swap-test.ts

```diff
@@ -0,0 +1,105 @@
+import { describe, expect, it } from "@jest/globals";
+
+import { swap } from "../swap";
+
+describe("swap", () => {
+  it.each([
+    {
+      arr: [1, 2, 3, 4],
+      i: 1,
+      j: 3,
+      expected: [1, 4, 3, 2],
+      description: "swaps two elements in an array of numbers",
+    },
+    {
+      arr: ["a", "b", "c", "d"],
+      i: 0,
+      j: 2,
+      expected: ["c", "b", "a", "d"],
+      description: "swaps two elements in an array of strings",
+    },
+    {
+      arr: [1, "a", true, Math.PI],
+      i: 1,
+      j: 3,
+      expected: [1, Math.PI, true, "a"],
+      description: "swaps two elements in an array of mixed types",
+    },
+    {
+      arr: [1, 2, 3],
+      i: 1,
+      j: 1,
+      expected: [1, 2, 3],
+      description: "does nothing when swapping the same index",
```

Comment:
Thanks for making this change -- sorry I wasn't entirely clear, I think the tests that are all testing the same thing make sense to group with an `it.each` but starting from about here the tests are a bit more unique. I would keep these as individual `it`s.

Apologies for the churn!